x-pack/plugins/observability/public/pages/cases/use_fetch_alert_detail.test.ts

```diff
@@ -0,0 +1,161 @@
+    'kibana.alert.rule.uuid': '69411af0-82a2-11ec-8139-c1568734434e',
+    'kibana.space_ids': ['default'],
+    'kibana.alert.rule.tags': [],
+    '@timestamp': '2022-01-31T18:20:57.204Z',
+    'kibana.alert.reason': 'Document count has reported no data over the past 1h for *',
+    'kibana.alert.duration.us': 13793555000,
+    'kibana.alert.instance.id': '*',
+    'kibana.alert.start': '2022-01-31T14:31:03.649Z',
+    'kibana.alert.uuid': '73c0d0cd-2df4-4550-862c-1d447e9c1db2',
+    'kibana.alert.status': 'active',
+    'kibana.alert.workflow_status': 'open',
+    'event.kind': 'signal',
+    'event.action': 'active',
+    'kibana.version': '8.1.0',
+    tags: [],
+  };
+
+  const id = '123';
+  const ruleType = createObservabilityRuleTypeRegistryMock();
+
+  beforeEach(() => {
+    mockUseKibanaReturnValue.services.http.get.mockImplementation(async () => getResult);
+    jest.spyOn(pluginContext, 'usePluginContext').mockImplementation(
+      () =>
+        ({
+          observabilityRuleTypeRegistry: ruleType,
+        } as unknown as PluginContextValue)
+    );
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('initially is not loading and does not have data', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, [boolean, TopAlert | null]>(() =>
+        useFetchAlertDetail(id)
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current).toEqual([false, null]);
+    });
+  });
+
+  it('returns no data when an error occurs', async () => {
+    mockUseKibanaReturnValue.services.http.get.mockImplementation(async () => {
+      throw new Error('an http error');
+    });
+
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, [boolean, TopAlert | null]>(() =>
+        useFetchAlertDetail('123')
+      );
+
+      await waitForNextUpdate();
+
+      expect(result.current).toEqual([false, null]);
+    });
+  });
+
+  it('retrieves the alert data', async () => {
+    await act(async () => {
+      const { result, waitForNextUpdate } = renderHook<string, [boolean, TopAlert | null]>(() =>
+        useFetchAlertDetail(id)
+      );
+
+      await waitForNextUpdate();
+      await waitForNextUpdate();
+
+      expect(result.current).toMatchInlineSnapshot(`
+        Array [
+          false,
+          Object {
+            "0": "a",
+            "1": " ",
+            "2": "r",
+            "3": "e",
+            "4": "a",
+            "5": "s",
+            "6": "o",
+            "7": "n",
+            "active": true,
+            "fields": Object {
+              "@timestamp": "2022-01-31T18:20:57.204Z",
+              "event.action": "active",
+              "event.kind": "signal",
+              "kibana.alert.duration.us": 13793555000,
+              "kibana.alert.instance.id": "*",
+              "kibana.alert.reason": "Document count has reported no data over the past 1h for *",
```

Comment:
Nit: Similar to my comment above `Document count reported no data in the last 1 hour for all hosts`.